src/common/snippets/src/pass/collapse_subgraph.cpp

```diff
@@ -190,18 +190,11 @@ auto is_supported_op(const std::shared_ptr<const Node> &n) -> bool {
 }
 
 auto has_supported_in_out(const std::shared_ptr<const Node> &n) -> bool {
-    auto supported = [&n](descriptor::Tensor& t) -> bool {
+    auto supported = [](descriptor::Tensor& t) -> bool {
         // TODO [122585] Need to add dynamic rank support
         if (t.get_partial_shape().rank().is_dynamic())
             return false;
-        // TODO [105804] int32 isn't supported in general because i32 emitters are required for bit-exact i32 calculations in some cases
-        //  So i32 is exclusively supported for specific set of operations
-        return TokenizeSnippets::get_supported_element_types().count(t.get_element_type()) != 0 ||
-                (t.get_element_type() == ov::element::i32 &&
-                        (ov::is_type<const opset1::Transpose>(n) ||
-                         ov::is_type<const opset1::Broadcast>(n) ||
-                         ov::is_type<const opset1::ReduceMax>(n) ||
-                         ov::is_type<const opset1::ReduceSum>(n)));
+        return true;
```

Comment:
Minor: We can simply this 😄 
`return t.get_partial_shape().rank().is_static();`

Reply:
Applied. Thanks Alexandra!